.github/workflows/build_and_test_pr.yml

```diff
@@ -0,0 +1,195 @@
+        -DCMAKE_OSX_DEPLOYMENT_TARGET:STRING=${{ matrix.macos_dev_target }}
+        -DLINK_WITH_PYTHON:BOOL=ON
+        -DPython_REQUIRED_VERSION:STRING=${{ steps.setup-python.outputs.python-version }}
+        -DPython_ROOT_DIR:PATH=$RUNNER_TOOL_CACHE/Python/${{ steps.setup-python.outputs.python-version }}/${{ matrix.python-arch }}/
+        -DBUILD_TESTING:BOOL=ON
+        -DBUILD_FORTRAN:BOOL=ON 
+        -DBUILD_PACKAGE:BOOL=OFF
+        -DDOCUMENTATION_BUILD:STRING=DoNotBuild
+        -DENABLE_OPENMP:BOOL=OFF
+        -DUSE_OpenMP:BOOL=OFF
+        ../
+
+    - name: Baseline Build
+      if: ${{ matrix.run_regressions }}
+      working-directory: ./baseline/build
+      # Building specific target list so that we don't waste time building the unit test binary
+      run: cmake --build . -j ${{ matrix.nproc }} --config Release --target ${{ matrix.targets }}
+
+    - name: Baseline Test
+      if: ${{ matrix.run_regressions }}
+      working-directory: ./baseline/build
+      run: ctest -C Release -E Basement -R integration -j 3  # TODO: Speed up basement so we don't have to skip it.
+
+    # BUILD AND TEST EVERYTHING ON THE CURRENT BRANCH
+
+    - name: Branch Checkout
+      uses: actions/checkout@v4
+      with:
+        path: branch
+
+    - name: Branch Create Build Directory
+      run: cmake -E make_directory ./branch/build/
+
+    - name: Branch Configure CMake
+      working-directory: ./branch/build
+      run: >
+        cmake 
+        -G "${{ matrix.generator }}"
+        -DCMAKE_BUILD_TYPE:STRING=Release
+        -DCMAKE_OSX_DEPLOYMENT_TARGET:STRING=${{ matrix.macos_dev_target }}
+        -DLINK_WITH_PYTHON:BOOL=ON
+        -DPython_REQUIRED_VERSION:STRING=${{ steps.setup-python.outputs.python-version }}
+        -DPython_ROOT_DIR:PATH=$RUNNER_TOOL_CACHE/Python/${{ steps.setup-python.outputs.python-version }}/${{ matrix.python-arch }}/
+        -DBUILD_TESTING:BOOL=ON
+        -DBUILD_FORTRAN:BOOL=ON
+        -DBUILD_PACKAGE:BOOL=OFF
+        -DDOCUMENTATION_BUILD:STRING=DoNotBuild
+        -DENABLE_OPENMP:BOOL=OFF
+        -DUSE_OpenMP:BOOL=OFF
+        ../
+
+    - name: Branch Build
+      id: branch_build
+      working-directory: ./branch/build
+      run: cmake --build . -j ${{ matrix.nproc }} --config Release
+
+    - name: Branch Test
+      working-directory: ./branch/build
+      run: ctest -C Release -E Basement -j 3
+
+    - name: Install Regression Tool
+      if: always() && matrix.run_regressions && steps.branch_build.outcome != 'failure'  # always run this step as long as we actually built
+      run: pip install energyplus-regressions
+
+    - name: Run Regressions
+      if: always() && matrix.run_regressions && steps.branch_build.outcome != 'failure'  # always run this step as long as we actually built
+      id: regressions
+      run: python ./branch/scripts/dev/gha_regressions.py ./baseline/build/testfiles ./branch/build/testfiles/ ${{ github.workspace }}/regressions
+
```

Comment:
And finally we report back to the dev in 3 ways:
 - Generating an artifact summary on the workflow run page that quickly summarizes the regression outcome.  This may not be necessary.  
![image](https://github.com/user-attachments/assets/286fc658-7122-4c7f-814e-2fe04680dcd8)
 - Uploading the full regression bundle as an artifact.  This will come down as a set of web pages in a zip.  Just extract it and open index.html, and enjoy all the diff information.  It's possible we could upload more than this...perhaps a Jupyter Notebook prepopulated with data for analyzing the diffs?
 - Finally we will also report back to the PR page to alert the dev of regressions.  If we choose to run on more than just PRs, then this will need to be modified to comment properly.  If you look at my commits here, you will see I tried to comment on commits, but realized it's probably not that important.